code/game/objects/structures/watercloset.dm

```diff
@@ -301,19 +301,21 @@
 				add_fingerprint(user)
 	else
 		if(I.is_wrench(user))
-			user.visible_message("<span class='warning'>[user] starts adjusting the bolts on \the [src].</span>", \
-								 "<span class='notice'>You start adjusting the bolts on \the [src].</span>")
-			playsound(src, 'sound/items/Ratchet.ogg', 100, 1)
-			if(do_after(user, src, 50))
-				if(anchored)
-					src.visible_message("<span class='warning'>[user] unbolts \the [src] from the floor.</span>", \
-								 "<span class='notice'>You unbolt \the [src] from the floor.</span>")
-					anchored = 0
-				else
-					src.visible_message("<span class='warning'>[user] bolts \the [src] to the floor.</span>", \
-								 "<span class='notice'>You bolt \the [src] to the floor.</span>")
-					anchored = 1
-
+			if(!on)
+				user.visible_message("<span class='warning'>[user] starts adjusting the bolts on \the [src].</span>", \
+									 "<span class='notice'>You start adjusting the bolts on \the [src].</span>")
+				playsound(src, 'sound/items/Ratchet.ogg', 100, 1)
+				if(do_after(user, src, 50))
+					if(anchored)
+						src.visible_message("<span class='warning'>[user] unbolts \the [src] from the floor.</span>", \
+									 "<span class='notice'>You unbolt \the [src] from the floor.</span>")
+						anchored = 0
```

Comment:
why don't simply add a `on = 0` on this line? Seems a little less obnoxious

Reply:
cuz update_icon
changed